### PR TITLE
Make `assign_datetime()` great again (fixing issues #101 and #113)

### PR DESCRIPTION
**Pull request by ramiromagno**
Thank you for your Pull Request! We have developed this task checklist from the
[Development Process
Guide](https://pharmaverse.github.io/sdtm.oak/CONTRIBUTING.html)
to help with the final steps of the process. Completing the below tasks helps to
ensure our reviewers can maximize their time on your code as well as making sure
the oak codebase remains robust and consistent.

Please check off each taskbox as an acknowledgment that you completed the task
or check off that it is not relevant to your Pull Request. This checklist is
part of the Github Action workflows and the Pull Request will not be merged into
the `devel` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull z
Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the 
[tidyverse style guide](https://style.tidyverse.org/). Run 
`styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should
consider realistic data scenarios and edge cases, e.g. empty datasets, errors,
boundary cases etc. - See 
[Unit Test Guide](https://pharmaverse.github.io/admiraldev/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you 
fully follow the 
[deprecation guidance](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples, including keywords
and families. Refer to the 
[categorization of functions](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the
`NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it
has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build oak site `pkgdown::build_site()` and check that all affected
examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Review by ShiyuC — APPROVED** [review]
Looks good to me

**Comment by rammprasad** [comment]
@ramiromagno - Tested with this example and seem not to be working. Am I missing something? I do not expect any problems in this scenario.

``` r
library(dplyr)
#> 
#> Attaching package: 'dplyr'
#> The following objects are masked from 'package:stats':
#> 
#>     filter, lag
#> The following objects are masked from 'package:base':
#> 
#>     intersect, setdiff, setequal, union
library(sdtm.oak)
# Read Specification
study_ct <- read.csv(system.file("raw_data/sdtm_ct.csv",
                                 package = "sdtm.oak"
))

# Read in raw data
vs_raw <- read.csv(system.file("raw_data/vitals_raw_data.csv",
                               package = "sdtm.oak"
)) %>%
  generate_oak_id_vars(
    pat_var = "PATNUM",
    raw_src = "vitals"
  )

# Map topic variable VSALL from raw variable ASMNTDN with the logic if ASMNTDN  == 1 then VSTESTCD = VSALL
vs_vsall <-
  hardcode_ct(
    raw_dat = condition_add(vs_raw, ASMNTDN == 1L),
    raw_var = "ASMNTDN",
    tgt_var = "VSTESTCD",
    tgt_val = "VSALL",
    ct_spec = study_ct,
    ct_clst = "C66741"
  ) %>%
  # Map VSDTC using assign_ct algorithm
  assign_datetime(
    raw_dat = vs_raw,
    raw_var = c("VTLD", "VTLTM"),
    tgt_var = "VSDTC",
    raw_fmt = c(list(c("d-m-y", "dd-mmm-yyyy")), "H:M")
  )

problems(vs_vsall$VSDTC)
#> # A tibble: 1 × 3
#>     ..i VTLD  VTLTM
#>   <int> <chr> <chr>
#> 1     4 ""    ""
```

<sup>Created on 2025-05-18 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>

**Comment by ramiromagno** [comment]
The `problems()` return value says it all, you have empty strings in your raw data.

I believe the approach has been to add a call with `admiral::convert_blanks_to_na()` before starting the derivations.

**Comment by rammprasad** [comment]
Thank you, @ramiromagno , Can we also ignore blanks? We do need NA_* or NA to be reported as problems.

**Comment by ramiromagno** [comment]
hey @rammprasad , thanks for your feedback.

Do I read you correctly, in that you are saying that you would rather have `NA` being reported as problems? Because that contradicts: https://github.com/pharmaverse/sdtm.oak/issues/113 and your own past remark in issue https://github.com/pharmaverse/sdtm.oak/issues/101:

> Expected Behavior:
>
>    1. I should be able to provide more than one format for a vector. As long as the date matches one of the formats, the function should process it.
>
>    2. Missing values should not be reported as problems.

**Comment by rammprasad** [comment]
Sorry, There was a type. I corrected it now- Can we also ignore blanks? We do NOT need NA_* or NA to be reported as problems.

**Comment by ramiromagno** [comment]
@rammprasad : No worries! Also, you have a typo in type :D (just kidding)

File: R/derive_blfl.R [hunk 1]
```diff
@@ -348,15 +383,23 @@ derive_blfl <- function(sdtm_in,
...
   )
 
+  # Assert that the input "DTC" column follows iso_8601 format
+  assertthat::assert_that(is_iso8601(sdtm_in[[domain_prefixed_names["dtc"]]]),
+    msg = paste(
+      domain_prefixed_names["dtc"],
+      "column does not follow ISO8601 format, please check."
+    )
+  )
+
```
Comment (rammprasad) [inline R/derive_blfl.R:393]: Can we add some details on why we had to update the derive_blfl function?

File: R/derive_blfl.R [hunk 2]
```diff
@@ -374,10 +417,10 @@ derive_blfl <- function(sdtm_in,
   }
 
   # Checking for columns of interest
-  con_col <- c(domain_prefixed_names[c("testcd", "dtc", "var_tpt")], "VISIT")
+  con_col <- c(domain_prefixed_names[c("testcd", "dtc", "tpt")], "VISIT")
```
Comment (rammprasad) [inline R/derive_blfl.R:420]: Similar comment. I would appreciate the context behind updating this.

**Review by rammprasad — COMMENTED** [review]
Working as expected. Added two minor comments,

File: R/derive_blfl.R (hunk 2 above)
Comment (ramiromagno) [reply R/derive_blfl.R:420]: See my previous comment. Idem.

**Review by rammprasad — APPROVED** [review]
